### PR TITLE
Handle PR edit

### PR DESCRIPTION
**Pull request by HeahDude**
What about this?

### CODE REVIEW

File: src/AppBundle/Subscriber/AutoLabelPRFromContentSubscriber.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ public function __construct(CachedLabelsApi $labelsApi)
     public function onPullRequest(GitHubEvent $event)
     {
         $data = $event->getData();
-        if ('opened' !== $action = $data['action']) {
+        if (!in_array($action = $data['action'], array('opened', 'edited'), true)) {
```
Comment (HeahDude) [inline src/AppBundle/Subscriber/AutoLabelPRFromContentSubscriber.php]: We could use a class to hold actions as constants like events. What do you think?

Comment (weaverryan) [inline src/AppBundle/Subscriber/AutoLabelPRFromContentSubscriber.php]: or a new value object for different types of events - e.g .`PullRequest` - with getters like `getAction()`, etc.

**Comment by weaverryan** [comment]
@HeahDude can you add a test that tests the addition/removal of labels - that's the only complex part of this change, and I do *not* want Carson making mistakes :)

**Comment by HeahDude** [comment]
@weaverryan Of courses, I thought about it and I'd love to! But I wanted your feedback first :)

Also, it's really hard to be sure about test I write since I cannot run them all... Could you please help me with this by adding some fixtures about an edited PR so I can work on that?

You could send a PR or push it so I can rebase, what do you think? Thanks!

File: src/AppBundle/Subscriber/AutoLabelPRFromContentSubscriber.php (hunk 1 above)
Comment (HeahDude) [inline src/AppBundle/Subscriber/AutoLabelPRFromContentSubscriber.php]: Yeah I thought about that when originally making a subscriber factory. But I really think we should keep the data array (payload) and maybe use some specific events like an `GitHubIssueEvent` which can hold specific method like `getAction` and fallback on the generic event in most case.

This should be the goal of another PR, what do you think?

**Comment by HeahDude** [comment]
I'm about to close this one, I've tested Carson with my own repo and it seems (the logs on github confirmed it) that we can't trust the payload concerning changes made to the title and description of any tickets :(

**Comment by yceruto** [comment]
> we can't trust the payload concerning changes made to the title and description of any tickets :(

Some news about this ? 

* https://developer.github.com/v3/activity/events/types/#pullrequestevent
* https://developer.github.com/v3/activity/events/types/#issuesevent

**Comment by HeahDude** [comment]
Thanks @yceruto for the doc, I'm aware of it ;) Sadly, AFAIK the payload is not safe regarding the change set (before/after), and in most on my real tests with my forks of Carson and projects the values in the payload was all about the old version.

I won't have time in the next weeks to look closer at this, feel free to test it on your side or take over this PR :)

**Comment by sstok** [comment]
You could issue (no pun) a new API call to get the new state, by getting the actual issue/pull-request. It would be nice if GitHub could provide both the old and new state in the payload 🤔

**Comment by HeahDude** [comment]
@sstok 

> You could issue (no pun) a new API call

I've thought about it but given the potential number of requests I've put that thought away.

> It would be nice if GitHub could provide both the old and new state in the payload

It does, it is just wrong most of the time with the new value of the changeset equal to the old one.

**Comment by stof** [comment]
> It does, it is just wrong most of the time with the new value of the changeset equal to the old

@HeahDude Have you reported this to Github ? It looks like a bug.
